test/spec/openapi/option.js

```diff
@@ -344,4 +344,154 @@ test('cache - yaml', t => {
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const openapiObject = fastify.swagger()
+    const schema = openapiObject.paths['/'].get.requestBody.content['application/json'].schema
+
+    t.ok(schema)
+    t.notOk(schema.properties.hello.examples)
+    t.equal(schema.properties.hello.example, 'world')
+  })
+})
+
+test('transforms examples in example if single object example', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, openapiOption)
+
+  const opts = {
+    schema: {
+      body: {
+        type: 'object',
+        required: ['hello'],
+        properties: {
+          hello: {
+            type: 'object',
+            properties: {
+              lorem: {
+                type: 'string'
+              }
+            },
+            examples: [{ lorem: 'ipsum' }]
+          }
+        }
+      }
+    }
+  }
+
+  fastify.get('/', opts, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const openapiObject = fastify.swagger()
+    const schema = openapiObject.paths['/'].get.requestBody.content['application/json'].schema
+
+    t.ok(schema)
+    t.notOk(schema.properties.hello.examples)
+    t.same(schema.properties.hello.example, { lorem: 'ipsum' })
+  })
+})
+
+test('uses examples if has multiple string examples', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, openapiOption)
+
+  const opts = {
+    schema: {
+      body: {
+        type: 'object',
+        required: ['hello'],
+        properties: {
+          hello: {
+            type: 'string',
+            examples: ['hello', 'world']
+          }
+        }
+      }
+    }
+  }
+
+  fastify.get('/', opts, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const openapiObject = fastify.swagger()
+    const schema = openapiObject.paths['/'].get.requestBody.content['application/json'].schema
+
+    t.ok(schema)
+    t.ok(schema.properties.hello.examples)
+    t.same(schema.properties.hello.examples, ['hello', 'world'])
+  })
+})
+
+test('uses examples if has multiple object examples', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.register(fastifySwagger, openapiOption)
+
+  const opts = {
+    schema: {
+      body: {
+        type: 'object',
+        required: ['hello'],
+        properties: {
+          hello: {
+            type: 'object',
+            properties: {
+              lorem: {
+                type: 'string'
+              }
+            },
+            examples: [{ lorem: 'ipsum' }, { hello: 'world' }]
+          }
+        }
+      }
+    }
+  }
+
+  fastify.get('/', opts, () => {})
+
+  fastify.ready(err => {
+    t.error(err)
+
+    const openapiObject = fastify.swagger()
+    const schema = openapiObject.paths['/'].get.requestBody.content['application/json'].schema
+
+    t.ok(schema)
+    t.ok(schema.properties.hello.examples)
+    t.same(schema.properties.hello.examples, [{ lorem: 'ipsum' }, { hello: 'world' }])
```

Comment:
https://swagger.io/docs/specification/adding-examples/
According to the document it should convert to something like
```js
{
  examples: {
    key: {
      value: { lorem: 'ipsum' }
    },
    secondKey: {
      value: { hello: 'world' }
    }
  }
}
```
for the example key here, maybe just make it random. We cannot specify the key from JSON schema.

Reply:
Totally missed this, to handle theses cases I created a function `convertExamplesArrayToObject` that converts these arrays to an object with all examples. Since I used a reduce to do this conversion, decided to use the prefix `key` + the value's `index` to build the keys.

The outputs for arrays and object examples were these (wrote tests for both cases)

### Array
Input
```js
examples: [['a', 'b', 'c'], ['d', 'f', 'g']]
```
Output
```js
{
  key0: {
    value: [ 'a', 'b', 'c' ]
  },
  key1: {
    value: [ 'd', 'f', 'g' ]
  }
}
```

### Object
Input
```js
examples: [{ lorem: 'ipsum' }, { hello: 'world' }]
```

Output
```js
{
  key0: {
    value: { lorem: 'ipsum' }
  },
  key1: {
    value: { hello: 'world' }
  }
}
```